core/src/main/scala/mongo4cats/bson/syntax.scala

```diff
@@ -31,10 +32,25 @@ trait CodecOps {
     def as[A: BsonDecoder]: Either[BsonDecodeError, A] = BsonDecoder[A].apply(value)
   }
 
-  implicit class BDocDecoderOps(val value: BsonDocument) {
+  implicit class BsonDocumentDecoderOps(val doc: BsonDocument) {
     def as[A: BsonDocumentDecoder]: Either[BsonDecodeError, A] =
-      BsonDocumentDecoder[A].apply(value)
+      BsonDocumentDecoder[A].apply(doc)
+
+    def getAs[A: BsonDecoder](k: String): Either[BsonDecodeError, A] =
+      if (doc.containsKey(k))
+        BsonDecoder[A]
+          .apply(doc.get(k))
+          .leftMap(err => BsonDecodeError(s"At key '$k': ${err.msg}"))
+      else BsonDecodeError(s"No key '$k' found").asLeft
+
+    def getOptional[A: BsonDecoder](k: String): Either[BsonDecodeError, Option[A]] =
+      if (doc.containsKey(k))
+        BsonDecoder[A]
+          .apply(doc.get(k))
+          .bimap(err => BsonDecodeError(s"At key '$k': ${err.msg}"), Some(_))
+      else Right(None)
```

Comment:
I could perhaps have done something such that there's an `Option[BsonValue]` kinda thing as a `BsonDecoder` instead of having a separate operation here, but I have occasionally found that style of definition problematic in the past because it can't accurately represent nested `Option`s.